graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -618,29 +552,55 @@ def _create_descriptor_from_property_definition(self, class_name, property_defin
     def _link_vertex_and_edge_types(self):
         """For each edge, link it to the vertex types it connects to each other."""
         for edge_class_name in self._edge_class_names:
-            edge_element = self._elements[edge_class_name]
-
-            if (EDGE_SOURCE_PROPERTY_NAME not in edge_element.properties or
-                    EDGE_DESTINATION_PROPERTY_NAME not in edge_element.properties):
-                if edge_element.abstract:
-                    continue
-                else:
-                    raise AssertionError(u'Found a non-abstract edge class with undefined '
-                                         u'endpoint types: {}'.format(edge_element))
+            edge = self._elements[edge_class_name]
 
-            from_class_name = edge_element.properties[EDGE_SOURCE_PROPERTY_NAME].qualifier
-            to_class_name = edge_element.properties[EDGE_DESTINATION_PROPERTY_NAME].qualifier
+            if edge.abstract:
+                continue
+            elif not(len(edge.in_connections) == 1 and len(edge.out_connections) == 1):
+                raise AssertionError(u'Found a non-abstract edge class with an incorrect number '
+                                     u'of endpoints: {}'.format(edge))
 
-            edge_schema_element = self._elements[edge_class_name]
+            from_class_name = next(iter(edge.in_connections))
```

Comment:
Let's still use the `get_only_element_from_collection` helper here, even though we did assert it above.